session/src/main/scala/akka/cassandra/session/CqlSessionProvider.scala

```diff
@@ -44,6 +49,75 @@ class DefaultSessionProvider(@unused system: ActorSystem, config: Config) extend
   }
 }
 
+/**
+ * ```
+ * akka {
+ *   discovery.method = config
+ * }
+ * akka.discovery.config.services = {
+ *   cassandra-service = {
+ *     endpoints = [
+ *       {
+ *         host = "127.0.0.1"
+ *         port = 9042
+ *       },
+ *       {
+ *         host = "127.0.0.2"
+ *         port = 9042
+ *       }
+ *     ]
+ *   }
+ * }
+ * my-cassandra-session {
+ *   session-provider = "akka.cassandra.session.AkkaDiscoverySessionProvider"
+ *   session-name = "my-session"
+ *   service {
+ *     name = "cassandra-service"
+ *     lookup-timeout = 1 s
+ *   }
+ * }
+ * ```
+ */
+class AkkaDiscoverySessionProvider(system: ActorSystem, config: Config) extends CqlSessionProvider {
+
+  /**
+   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
+   */
+  private def readNodes(serviceName: String, lookupTimeout: FiniteDuration)(
+      implicit system: ActorSystem): Future[immutable.Seq[String]] = {
+    import system.dispatcher
+    val discovery = Discovery(system).discovery
+    discovery.lookup(serviceName, lookupTimeout).map { resolved =>
+      resolved.addresses.map(addr => addr.host + ":" + addr.port)
+    }
+  }
+
+  /**
+   * Expect a `service` section in Config and use Akka Discovery to read the addresses for `name` within `lookup-timeout`.
+   */
+  private def readNodes(config: Config)(implicit system: ActorSystem): Future[immutable.Seq[String]] =
+    if (config.hasPath("service")) {
+      val serviceName = config.getString("service.name")
+      val lookupTimeout = config.getDuration("service.lookup-timeout").asScala
+      readNodes(serviceName, lookupTimeout)
+    } else throw new IllegalArgumentException(s"config $config does not contain `service` section")
```

Comment:
toString of `config` might not look nice (and we don't want \n in such messages)
isn't error message of ordinary `config.getConfig("service")` good enough?